app/mailers/payment_mailer.rb

```diff
@@ -1,9 +1,9 @@
 class PaymentMailer < ApplicationMailer
   helper :application
 
-  def confirmation(payment, payment_date_timestamp)
+  def confirmation(payment)
     @payment = payment
-    @payment_date = Time.at(payment_date_timestamp).to_date
+    @payment_date = payment.payroll_run.payment_date
```

Comment:
How about we encapsulate this by adding a delegated `Payment#payment_date` (or rather `Payment#scheduled_payment_date`) method?